src/NHibernate/Hql/Ast/ANTLR/Tree/FromElementType.cs

```diff
@@ -426,21 +426,13 @@ public string[] ToColumns(string tableAlias, string path, bool inSelect, bool fo
 					//              B) otherwise, we need to use the persister's
 					//                  table name as the column qualification
 					//      2) otherwise (not correlated), use the given alias
-					if (IsCorrelation)
+					bool safeUseAlias = _fromElement.Walker.IsComparativeExpressionClause || IsMultiTable;
```

Comment:
What does `safeUseAlias` name mean?

Reply:
Means table alias can be used without any additional checks...

Reply:
Hm.. We shouldn't use table aliases for subqueries in UPDATE statement...